drivers/syslog/ramlog.c

```diff
@@ -125,8 +125,8 @@ static const struct file_operations g_ramlogfops =
 
 #ifdef CONFIG_RAMLOG_SYSLOG
 static char g_sysbuffer[CONFIG_RAMLOG_BUFSIZE]
-#ifdef CONFIG_RAMLOG_BUFFER_SECTION
-                               locate_data(CONFIG_RAMLOG_BUFFER_SECTION)
+#ifdef RAMLOG_BUFFER_SECTION
```

Comment:
Why not use the #define in the preprocessor section to define the "locate" as was done in the other PR?

Reply:
Since RAMLOG_BUFFER_SECTION and IOB_SECTION are defined by Make.defs:
https://github.com/apache/incubator-nuttx/pull/6894/files#diff-803dc937d614cd425d920dbd26bcf947ea598029928167af68fae122b89718deR41
https://github.com/apache/incubator-nuttx/pull/6894/files#diff-30811896863304a4ba8b467eebd5a110e46e8f84b311deb349c0640c9187d7b2R42

Reply:
I suggesting using the pattern (choose the naming as you like)

```
#if defined(RAMLOG_BUFFER_SECTION)
#  define RAM_LOG_LOCATE locate_data(RAMLOG_BUFFER_SECTION)
# else
#  define RAM_LOG_LOCATE
#endif  
```
...
```
static char g_sysbuffer[CONFIG_RAMLOG_BUFSIZE] RAM_LOG_LOCATE
```

Reply:
Since this macro just reference once, I prefer to #ifdef/#else/#endif directly instead introducing a new macro. Anyway, I change g_sysbuffer definition to ensure both places use the same style.